lib/Controller/KeyController.php

```diff
@@ -205,6 +205,26 @@ public function createPublicKey(string $csr): DataResponse {
 		return new DataResponse(['public-key' => $publicKey]);
 	}
 
+	/**
+	 * Set public key
+	 *
+	 * @NoAdminRequired
```

Comment:
The requirements are really that a user should be able to set its own public key? And not that an admin should be able to set a user's public key?

Reply:
that should be done by users as there may be situation where the user's certificate is issued by a different component from the Nextcloud server app and the user should be able to upload their certificate to enable end-to-end encrypted share feature
I would guess that we want logged-in user to be able to do it as if they were uploading just a certificate signing request (this workflow works without administrator doing something)

Reply:
>  I would guess that we want logged-in users to be able to do it as if they were uploading just a certificate signing request (this workflow works without administrator doing something)

Please make sure of it. If this is true, then it looks ok :).

Reply:
sorry if that is unclear
yes, normal users need to be able to upload their certificate to enable sharing
now that you ask for, I think we should ensure the user certificate is valid before storing it